src/module.c

```diff
@@ -419,14 +419,15 @@ int MGraph_Explain(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
     if (ast->indexNode != NULL) { // index operation
         const char *strPlan = Index_OpPrint(ast->indexNode);
         RedisModule_ReplyWithSimpleString(ctx, strPlan);
-        return REDISMODULE_OK;
+        goto cleanup;
     }
 
     ExecutionPlan *plan = NewExecutionPlan(ctx, g, graph_name, ast, true);
     char* strPlan = ExecutionPlanPrint(plan);
     RedisModule_ReplyWithStringBuffer(ctx, strPlan, strlen(strPlan));
-    
     ExecutionPlanFree(plan);
+
+cleanup:
```

Comment:
ExecutionPlanFree should be under cleanup
ExecutionPlanFree should handle NULL plan.